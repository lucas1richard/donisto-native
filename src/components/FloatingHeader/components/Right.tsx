import * as React from 'react';
import Touchable from 'components/Touchable';
import { View, StyleSheet } from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';
import theme from 'theme/iftheme';

const hitSlop = {
  top: 15,
  bottom: 15,
  left: 50,
  right: 50
};

function Right({ noCloseOption, onClose }: {
  noCloseOption?: boolean,
  blue?: boolean,
  style?: Object,
  onClose: () => void
}) {
  if (noCloseOption) {
    return <View style={styles.wrapper} />;
  }

  return (
    <Touchable onPress={onClose} hitSlop={hitSlop}>
      <View style={styles.wrapper}>
        <MaterialIcons
          size={30}
          name="close"
          style={{ textAlign: 'center' }}
          color={theme.blue}
        />
      </View>
    </Touchable>
  );
}

export default Right;

const styles = StyleSheet.create({
  wrapper: {
    position: 'absolute',
    top: 12,
    right: 12,
    // backgroundColor: 'rgba(0,0,0,0.1)',
    height: 34,
    width: 34,
    padding: 2,
    borderRadius: 17,
    zIndex: 100
  }
});
