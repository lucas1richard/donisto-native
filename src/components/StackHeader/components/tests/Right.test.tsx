import * as React from 'react';
import { shallow } from 'enzyme';
import Right, { IRightProps } from '../Right';

describe('<Right />', () => {
  const renderComponent = (props: IRightProps) => shallow(<Right {...props} />);
  it('renders', () => {
    expect(renderComponent({
      onClose: jest.fn()
    })).toMatchSnapshot();
  });
});