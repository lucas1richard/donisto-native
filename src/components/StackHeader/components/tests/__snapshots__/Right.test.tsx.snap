// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Right /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Right
    onClose={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <Component
        style={
          Object {
            "borderRadius": 17,
            "height": 34,
            "padding": 2,
            "width": 34,
            "zIndex": 100,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="#0F2243"
          name="close"
          size={30}
          style={
            Object {
              "textAlign": "center",
            }
          }
        />
      </Component>,
      "hitSlop": Object {
        "bottom": 15,
        "left": 50,
        "right": 50,
        "top": 15,
      },
      "onPress": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Icon
          allowFontScaling={false}
          color="#0F2243"
          name="close"
          size={30}
          style={
            Object {
              "textAlign": "center",
            }
          }
        />,
        "style": Object {
          "borderRadius": 17,
          "height": 34,
          "padding": 2,
          "width": 34,
          "zIndex": 100,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": false,
          "color": "#0F2243",
          "name": "close",
          "size": 30,
          "style": Object {
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <Component
          style={
            Object {
              "borderRadius": 17,
              "height": 34,
              "padding": 2,
              "width": 34,
              "zIndex": 100,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            color="#0F2243"
            name="close"
            size={30}
            style={
              Object {
                "textAlign": "center",
              }
            }
          />
        </Component>,
        "hitSlop": Object {
          "bottom": 15,
          "left": 50,
          "right": 50,
          "top": 15,
        },
        "onPress": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Icon
            allowFontScaling={false}
            color="#0F2243"
            name="close"
            size={30}
            style={
              Object {
                "textAlign": "center",
              }
            }
          />,
          "style": Object {
            "borderRadius": 17,
            "height": 34,
            "padding": 2,
            "width": 34,
            "zIndex": 100,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": false,
            "color": "#0F2243",
            "name": "close",
            "size": 30,
            "style": Object {
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
